src/main/java/org/jabref/logic/importer/fetcher/IsbnFetcher.java

```diff
@@ -59,6 +62,13 @@ public String getName() {
             bibEntry = isbnViaOttoBibFetcher.performSearchById(identifier);
         }
 
+        // nothing found at ebook.de and ottobib
+        if (!bibEntry.isPresent()) {
+            FXDialog noEntryFoundDialog = new FXDialog(AlertType.ERROR);
```

Comment:
This is not the right place. You cannot put GUI code into logic/model classes.
See https://devdocs.jabref.org/getting-into-the-code/high-level-documentation

The right place to check if the results are empty would be here: Than you have it generic as well for all kind of fetchers
https://github.com/JabRef/jabref/blob/7f4c36c03629498db8e56281906241a82846e1e3/src/main/java/org/jabref/gui/EntryTypeViewModel.java#L147

Reply:
oops, my mistake. I'll fix it later today. I have another question about the ottobib fetcher though, should getting no results throw an exception? Ottobib seems like a special case since supplying an ISBN that leads to no entry doesn't give a 404 but instead leads to a special page with an error message.

Reply:
I thought about throwing a fetcher exception as well. But on the other hand we have optional for those cases.
So I would not throw an exception.
@tobiasdiez

Reply:
Thanks for your work so far.

Exceptions are only for exceptional program states. An unkown ISBN is probably a regular case in Ottobib. Thats probably why Ottobib does not show 404, but a page saying: unkown ISBN.
See: https://github.com/HugoMatilla/Effective-JAVA-Summary#57-use-exceptions-only-for-exceptional-conditions